GeoLib/Surface.cpp

```diff
@@ -50,17 +52,30 @@ void Surface::addTriangle (std::size_t pnt_a, std::size_t pnt_b, std::size_t pnt
 	if (pnt_a == pnt_b || pnt_a == pnt_c || pnt_b == pnt_c)
 		return;
 
-	_sfc_triangles.push_back (new Triangle(_sfc_pnts, pnt_a, pnt_b, pnt_c));
+	_sfc_triangles.push_back(new Triangle(_sfc_pnts, pnt_a, pnt_b, pnt_c));
 	if (!_bounding_volume) {
 		std::vector<size_t> ids(3);
 		ids[0] = pnt_a;
 		ids[1] = pnt_b;
 		ids[2] = pnt_c;
 		_bounding_volume = new AABB<Point>(_sfc_pnts, ids);
+		if (_surface_grid == nullptr) {
+			_surface_grid = new SurfaceGrid(this);
+	}
 	} else {
-		_bounding_volume->update (*_sfc_pnts[pnt_a]);
-		_bounding_volume->update (*_sfc_pnts[pnt_b]);
-		_bounding_volume->update (*_sfc_pnts[pnt_c]);
+		if (_bounding_volume->update(*_sfc_pnts[pnt_a])
+			|| _bounding_volume->update(*_sfc_pnts[pnt_b])
```

Comment:
To the second part: `|` is bit-wise-or. I think a logic-or, which is `||`, is correct here.